### PR TITLE
DS-4466 REST API should obtain language settings from Client

### PR DESCRIPTION
**Pull request by ddinuzzo**
## References
* [DS-4466](https://jira.lyrasis.org/browse/DS-4466)
* [Approved REST Contract](https://github.com/DSpace/Rest7Contract/pull/122)

## Description
This PR add the general infrastructure needed to deal with the accept-language header from the client and test specific to proof that the submission definition are managed properly in a multi-lang environment. The implementation of the multilanguage support in the authority control is left outside as a major rework of this aspects is undergoing.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/master/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/SubmissionFormRestRepository.java [hunk 1]
```diff
@@ -65,4 +85,13 @@ public SubmissionFormRest findOne(Context context, String submitName) {
     public Class<SubmissionFormRest> getDomainClass() {
         return SubmissionFormRest.class;
     }
+
+    public void reload() throws DCInputsReaderException {
```
Comment (tdonohue) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/SubmissionFormRestRepository.java:89]: Can we add JavaDocs here?  It looks like (to me) this method should say something like:  `Reload the current Submission Form configuration based on the currently supported locales. This method can be used to force a reload if the configured supported locales change.`

**Review by tdonohue — APPROVED** [review]
👍 Overall, the code here looks great @ddinuzzo !  The tests look good as well & do a good job of providing this works as expected for language-specific submission configurations (which is one of the main use cases).  That said,  I do have one minor request for JavaDocs inline below.  Please update those JavaDocs before we can merge this.

**Comment by abollini** [comment]
@tdonohue this should be ready to go now

**Comment by benbosman** [comment]
@ddinuzzo It seems there's no IT yet for the use case of multiple languages in header

Can you add an IT with default locale English and the language header to `fr;q=1, it;q=0.9`
This should try French (but not find it), and go for the second option Italian (and not go to the default English)

**Review by benbosman — APPROVED** [review]
Thanks @ddinuzzo 
This looks good to me, I only found one use case that's not supported
If I set `webui.supported.locales = en, de, nl` and I have submission forms in `nl` but not in `de`, and I use `Accept-Language: de;q=0.9,nl;q=0.8,en;q=0.7`, this returns the English form (the default)
This is because it's assumed that all webui.supported.locales are available for each submission form, and `I18nUtil.isSupportedLocale(current)` doesn't take into account whether there's a submission form for this language.

I don't see this as a blocking issue, but I would prefer a github issue would be created for this

**Comment by tdonohue** [comment]
This is at +2, merging

**Pull request merged**